services/user-service.js

```diff
@@ -116,6 +118,102 @@ exports.signUp = async function (username, unhashed_password, last_name, first_n
     }
 }
 
+exports.updatePatientInfo = async function (userid, interests, biography) {
+  
+  try {
+    //delete interests:
+    let patient = await User.findOne({where: {
+        id: userid
+    }});
+    await PatientXAttribute.destroy({
+        where: {
+            patient_id: userid
+        }
+    });
+    let userInterests = [];
+    for (i = 0; i < interests.length; i++) {
+        userInterests.push({
+            patient_id: userid,
+            attribute_id: interests[i]
+        });
+    }
+    await PatientXAttribute.bulkCreate(userInterests);
+
+    //create new patient info
+    let patientInfo = await PatientInfo.build({biography});
+    await patient.setPatientInfo(patientInfo);
+    let createdInfo = await patient.save();
+    return createdInfo;
+  } catch (e) {
+      if (e instanceof Sequelize.ValidationError) {
+          let errorMessage = "The following values are invalid:";
+          e.errors.forEach((error) => {
+              errorMessage += `\n${error.path}: ${error.message}`;
+          });
+          throw new BadRequestException(errorMessage);
+      }
+      throw new InternalErrorException("A problem occurred when updating user info", e);
+  }
+}
+
+exports.createInterests = async function (userid, interests) {

```

Comment:
also, apparently we aren't using these helper methods, so we should probably delete them too